core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -624,6 +624,15 @@ public IRubyObject rewind() {
         return this;
     }
 
+    @JRubyMethod(name = "empty?", meta = true)
+    public static IRubyObject empty_p(ThreadContext context, IRubyObject recv, IRubyObject arg) {
+        Ruby runtime = context.runtime;
+        RubyString path = StringSupport.checkEmbeddedNulls(runtime, RubyFile.get_path(context, arg));
+        RubyFileStat fileStat = runtime.newFileStat(path.asJavaString(), false);
+        boolean isDirectory = fileStat.directory_p().isTrue();
+        return runtime.newBoolean(isDirectory && entries19(context, recv, arg).getLength() <= 2);
```

Comment:
The check for `<= 2` looks a little odd, but that is because `entries19` includes the paths "." and "..". I couldn't find functionality that provides the "real" paths in the directory.

Let me know if you know of a clearer way to do this check.